native/src/widget/pane_grid.rs

```diff
@@ -119,22 +119,37 @@ where
     /// Creates a [`PaneGrid`] with the given [`State`] and view function.
     ///
     /// The view function will be called to display each [`Pane`] present in the
-    /// [`State`].
+    /// [`State`]. [`bool`] is set if the pane is maximized.
     pub fn new<T>(
         state: &'a State<T>,
-        view: impl Fn(Pane, &'a T) -> Content<'a, Message, Renderer>,
+        view: impl Fn(Pane, &'a T, bool) -> Content<'a, Message, Renderer>,
     ) -> Self {
-        let elements = {
-            state
-                .panes
-                .iter()
-                .map(|(pane, pane_state)| (*pane, view(*pane, pane_state)))
-                .collect()
+        let (contents, state) = if let Some((pane, pane_state)) =
+            state.maximized.and_then(|pane| {
+                state.panes.get(&pane).map(|pane_state| (pane, pane_state))
+            }) {
+            (
+                Contents::Maximized(pane, view(pane, pane_state, true)),
+                state::Scoped::Maximized(Node::Pane(pane)),
```

Comment:
These seem inherently coupled, but the current data model allows us to potentially have desync. Could we simply merge `Scoped` with `Contents`?

Reply:
:+1: Good call, fixed in 7de9d24